### ISSUE TITLE
New CLI: --no-appearance and xml query error

### ISSUE THREAD

**Issue by BWibo** [3dcitydb/importer-exporter#165]
I just came accross a possible issue using the new CLI:

I am trying to do a CityGML export using the following XMLQuery with a spatial and a property filter criteria:

```xml
<query xmlns="http://www.3dcitydb.org/importer-exporter/config">
  <typeNames>
    <typeName xmlns:bldg="http://www.opengis.net/citygml/building/2.0">bldg:Building</typeName>
  </typeNames>
  <filter>
    <and>
      <intersects>
        <valueReference>bldg:boundedBy/bldg:GroundSurface/bldg:lod2MultiSurface</valueReference>
        <polygon srid="4326">
          <exterior>-73.84792 40.87134 -73.84725 40.870987 -73.84699 40.870853 ... ... ...</exterior>
        </polygon>
      </intersects>
      <propertyIsEqualTo>
        <valueReference>bldg:function</valueReference>
        <literal>1111</literal>
      </propertyIsEqualTo>
    </and>
  </filter>
</query>
```

When I add the `--no-appearance` option to the CLI call, the filter criteria of the xml query are ignored and the entire database is exported. When I remove the `--no-appearance` option, the XML filter criteria work as expected.

Log of the Docker image I used, based on 53643e7432c5db8818f994794f955a0870bb06af:
```text
# 3DCityDB ImporterExporter Docker #############################################
################################################################################
# Official repo   github.com/3dcitydb/importer-exporter
# Docker repo     github.com/tum-gis/3dcitydb-importer-exporter-docker

# Version info -----------------------------------------------------------------
  sha1 short    53643e7
  sha1          53643e7432c5db8818f994794f955a0870bb06af
  commit time   Wed Jan 13 12:12:44 2021 +0100
  branch        master

...
...

Command line passed to 3DCityDB ImporterExporter:
        export -H ssd -d nyc_bldg -u postgres --no-appearance -o /share/data/exports/fme_output/withFunction/Bronx_Allerton-Pelham Gardens.xml @/share/data/fme_output/withFunction/Bronx_Allerton-Pelham Gardens.xml

[13:54:38 INFO] Starting 3D City Database Importer/Exporter, version 4.3.0
[13:54:38 INFO] Loading plugins
[13:54:38 INFO] Initializing plugin org.citydb.plugins.spreadsheet_gen.SPSHGPlugin
[13:54:38 INFO] Loading database schema mapping
[13:54:39 INFO] Loading ADE extensions
[13:54:39 INFO] Loading CityGML and ADE contexts
[13:54:39 INFO] Initializing application environment
[13:54:39 INFO] Executing 'export' command
[13:54:39 INFO] Connecting to database 'postgres@ssd:5432/nyc_bldg'.
[13:54:40 INFO] Database connection established.
[13:54:40 INFO] 3D City Database: 4.0.3
[13:54:40 INFO] DBMS: PostgreSQL 11.10 (Ubuntu 11.10-1.pgdg18.04+1)
[13:54:40 INFO] Connection: postgres@ssd:5432/nyc_bldg
[13:54:40 INFO] Schema: citydb
[13:54:40 INFO] SRID: 32118 (Projected)
[13:54:40 INFO] SRS: NAD83 / New York Long Island
[13:54:40 INFO] gml:srsName: urn:ogc:def:crs:EPSG::32118
[13:54:40 INFO] Spatial indexes are enabled.
[13:54:40 INFO] Normal indexes are enabled.
[13:54:40 INFO] Exporting to file: /share/data/exports/fme_output/withFunction/Bronx_Allerton-Pelham Gardens.xml
^C^C (Manually canceled as the output file grew to >> 5GB)

```
Log output without `--no--appearance`:

```text
...
...
...

Command line passed to 3DCityDB ImporterExporter:
        export -H ssd -d nyc_bldg -u postgres -o /share/data/exports/fme_output/withFunction/Bronx_Allerton-Pelham Gardens.xml @/share/data/fme_output/withFunction/Bronx_Allerton-Pelham Gardens.xml


[13:59:46 INFO] Starting 3D City Database Importer/Exporter, version 4.3.0
[13:59:46 INFO] Loading plugins
[13:59:46 INFO] Initializing plugin org.citydb.plugins.spreadsheet_gen.SPSHGPlugin
[13:59:46 INFO] Loading database schema mapping
[13:59:46 INFO] Loading ADE extensions
[13:59:46 INFO] Loading CityGML and ADE contexts
[13:59:47 INFO] Initializing application environment
[13:59:47 INFO] Executing 'export' command
[13:59:47 INFO] Connecting to database 'postgres@ssd:5432/nyc_bldg'.
[13:59:47 INFO] Database connection established.
[13:59:47 INFO] 3D City Database: 4.0.3
[13:59:47 INFO] DBMS: PostgreSQL 11.10 (Ubuntu 11.10-1.pgdg18.04+1)
[13:59:47 INFO] Connection: postgres@ssd:5432/nyc_bldg
[13:59:47 INFO] Schema: citydb
[13:59:47 INFO] SRID: 32118 (Projected)
[13:59:47 INFO] SRS: NAD83 / New York Long Island
[13:59:47 INFO] gml:srsName: urn:ogc:def:crs:EPSG::32118
[13:59:47 INFO] Spatial indexes are enabled.
[13:59:47 INFO] Normal indexes are enabled.
[13:59:48 INFO] Created texture files folder 'appearance'.
[13:59:48 INFO] Exporting to file: /share/data/exports/fme_output/withFunction/Bronx_Allerton-Pelham Gardens.xml
[14:00:01 INFO] Cleaning temporary cache.
[14:00:01 INFO] Exported city objects:
[14:00:01 INFO] bldg:Building: 4126
[14:00:01 INFO] bldg:GroundSurface: 4126
[14:00:01 INFO] bldg:RoofSurface: 5345
[14:00:01 INFO] bldg:WallSurface: 35253
[14:00:01 INFO] Processed geometry objects: 134172
[14:00:01 INFO] Total export time: 13 s.
[14:00:01 INFO] Database export successfully finished.
```

**Comment by clausnagel** [comment]
Thanks for spotting this, @BWibo. Will work on a fix.

**Comment by clausnagel** [comment]
Fixed with 442369621988427e0963ffc4581d1e59a1343dbe.

**Comment by BWibo** [comment]
Thx for the fixed. Just tested with a fresh build form 3ae3a2f and it appears to be working now.
